client/src/sections/resources/Banishers.tsx

```diff
@@ -0,0 +1,171 @@
+import React from "react";
+import Line from "../../components/Line";
+import Tile from "../../components/Tile";
+import { useHaveEquipped, useMyTurncount } from "../../hooks/useCall";
+import useGet from "../../hooks/useGet";
+import useHave from "../../hooks/useHave";
+import { returnBanishes } from "../../util/chunkBanish";
+import { $item, $skill } from "../../util/makeValue";
+import { plural } from "../../util/text";
+
+/**
+ * Class used to store banishers. Unless otherwise specified, all params are strings.
+ * @param name The name of the banisher source
+ * @param numberRemaining Number of the banishers remaining (#)
+ * @param access Boolean representing if the user can access the banish
+ * @param banishType Must be "skill", "item", or "equip"; kind of banish
+ * @param href (optional) The target for a link
+ * @param isEquipped (optional) Boolean of if the item needed is equipped
+ * @param equipItem (optional) String of the item you need to equip
+ * @param extraString (optional) An extra string to append to the end of the tile
+ */
+class BanisherObject {
+  name: string;
+  numberRemaining: number;
+  access: boolean;
+  banishType: string;
+  href: string;
+  banishedMonster: string;
+  isEquipped: boolean;
+  equipItem: string;
+  extraString: string;
+
+  constructor(
+    name: string,
+    numberRemaining: number,
+    access: boolean,
+    banishType: "skill" | "item" | "equip",
+    href?: string,
+    banishedMonster?: string,
+    isEquipped?: boolean,
+    equipItem?: string,
+    extraString?: string
+  ) {
+    this.name = name;
+    this.numberRemaining = numberRemaining;
+    this.access = access;
+    this.banishType = banishType;
+    this.banishedMonster = banishedMonster ?? "";
+    this.href = href ?? "";
+    this.isEquipped = isEquipped ?? true;
+    this.equipItem = equipItem ?? "";
+    this.extraString = extraString ?? "";
+  }
+
+  // Returns if the banish is accessible for filtering.
+  accessible(): boolean {
+    return this.access && this.numberRemaining > 0;
+  }
+
+  // Renders the tile. Switches the line's stylings based on banishType.
+  render(): React.ReactNode {
+    const possibleStrings: { [key: string]: string } = {
```

Comment:
give this a type of `{ item: string, skill: string, equip: string }`, so that `keyof typeof possibleStrings` matches with the possible values of `banishType`